src/main/resources/com/rultor/agents/req/merge.sh

```diff
@@ -20,6 +20,12 @@ while [ $(git show-branch "${BRANCH}" 2>/dev/null | wc -l) -gt 0 ]; do
 done
 
 git checkout -B "${BRANCH}" "fork/${fork_branch}"
+git fetch origin
+if git log --name-status --oneline HEAD..origin/${head_branch} | grep ".rultor.yml"; then
+  echo "Since original repository has different version of .rultor.yml file you have to sync your fork with the upstream repository."
+  exit 1
+fi
```

Comment:
@xupyprmv I know that the current architecture cannot properly do this, we need to make a change obviously :) 
The problem with requiring `sync fork with upstream` is that this will inevitably cause builds that worked fine before to fail. We cannot add a requirement to sync your `.rultor.yml` with `master` to Rultor, just because it's harder to implement this properly on the Rultor end.

Also, I mean we're both clear on how to do this now I think. We get the initial profile from master, use it to answer the GitHub request.
Then get the merged profile and use it for the build itself, plus check that it doesn't adjust  information about who's commander for what and such.

So basically the three things we need are:
* some functionality to fetch the merged profile.
* some functionality to verify it does not contain changes to authentication things
* adjust the actual build to run on the merged profile

=> my suggestion to adjust the solution here would be this:
* create puzzles for the above three things where they are missing
* remove changes to the bash script
* Fix whatever @mkordas finds

=> we're good :)